### DISCUSSION TITLE
assigning a voltage excitation

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4414] [Q&A] [answered]
Hi, I want to assign a voltage excitation in ICEPAK and Maxwell3D multiphysics field coupling analysis,the  solution type of Maxwell3D is "DC Conduction". 
I've  these parameters as follows.

    sheet_name = "V_0"
    m3d.assign_voltage(sheet_name, amplitude=0)

or

    face_id = m3d.modeler[sheet_name].id
    m3d.assign_voltage(face_id, amplitude=0)

or

    face_id = m3d.modeler[sheet_name].id
    m3d.assign_voltage([face_id], amplitude=0)

However, the above three methods  result in the following error message

  [error] script macro error: excitation 'voltage_mf5f9g': assignment must contain exactly one selection.

Looking forward to your help！

**Comment by gmalinve** [comment, marked as answer]
Hi @aknsj,

I was able to reproduce your issue if the object name is passed to the method but not in the other two cases.
Are you able to provide the entire script that fails?
in case you are interested this is the PR where I fixed this: #4433

Giulia
